### PR TITLE
Refactor code for 'server show', 'server create' output

### PR DESCRIPTION
**Pull request by lidl**
This set of changes incorporates the refactoring of the 'server show' output
code into a new helper file, and changes both 'server show' and the out
from 'server create' to use the now separated out code.

### CODE REVIEW

File: lib/chef/knife/bmcs_server_show.rb [hunk 1]
```diff
@@ -103,10 +38,10 @@ def run
             raise unless service_error.serviceCode == 'NotAuthorizedOrNotFound'
           else
             add_vnic_details(vnic_info.data)
-            # for now, only display information for primary vnic
             if vnic_info.data.is_primary == true
+              vnic_array.unshift(vnic_info.data) # make primary interface first in the array
+            else
```
Comment (briangustafson) [inline lib/chef/knife/bmcs_server_show.rb]: Does this now show secondary vnics? Let's put that in another commit so that we can merge this now.

**Comment by lidl** [comment]
I will update this pull request with an updated one, once the multi-vnic pull request has
landed.  It's a hassle to keep rebasing this changeset since it's built on top of the changes
in the multi-vnic branch.

**Comment by lidl** [comment]
In case it wasn't obvious (due to the preservation of the old code commit dates), this pull
request has been updated with an up-to-date after the name change set of patches.
